### PR TITLE
Introduce PHPStorm Tools

### PR DESCRIPTION
**Pull request by gerner-spryker**
## PR Description
PHPStorm IDE supporting tools have been released. Documentation is available outlining the steps to enable this support and providing details on the features and functionality offered.

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Currently, the installation script is supported only on Mac and Linux systems.
+It is recommended to rerun this command periodically to ensure that the latest updates are applied.
+
+The following commands are included:
+- `console code:sniff:style -f`: Performs style fixes.
+- `console code:phpstan`: Executes static analysis.
+- `console code:sniff:architecture`: Checks for architecture rule violations.
+Each of these commands can be run at either the module or project level.
+
+Within PHPStorm, they are accessible via the External Tools panel, which allows for execution directly from the project directory tree by right-clicking on the desired directory.
```
[the remark this answers is not in the document]
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:27]: yeah, i was thinking the same 👍

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+It is recommended to rerun this command periodically to ensure that the latest updates are applied.
+
+The following commands are included:
+- `console code:sniff:style -f`: Performs style fixes.
+- `console code:phpstan`: Executes static analysis.
+- `console code:sniff:architecture`: Checks for architecture rule violations.
+Each of these commands can be run at either the module or project level.
+
+
+
```
Comment (gerner-spryker) [inline docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:28]: @andriitserkovnyi Pleaes add here an image 
image

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+related:
+  - title: Code sniffer
+    link: docs/scos/dev/sdk/development-tools/code-sniffer.html
+  - title: PHPStan
+    link: docs/scos/dev/sdk/development-tools/phpstan.html
+
+---
+
+## PHPStorm Support
+Spryker provides support for tools that perform code style fixes, static analysis, and architecture checks, each accessible within PHPStorm’s External Tools panel. These tools are available at both module and project levels and can be executed directly by right-clicking within the project directory tree.
```
[the remark this answers is not in the document]
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:17]: Sorry this is customer oriented. I really meant here, module level or the entire project level.

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+The following tools are included:
+- `console code:sniff:style -f`: Applies coding style fixes.
+- `console code:phpstan`: Runs static analysis.
+- `console code:sniff:architecture`: Detects architecture rule violations.
+
+### Installation
+
+**Prerequisite**: Ensure that `awk` is installed on your local machine, as it is required to run the commands directly from PHPStorm.
```
[the remark this answers is not in the document]
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:26]: Too many conditions, Mac or Linux, what package manager you use... I think it's too much.

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+To install the Spryker development tools configuration in PHPStorm:
```
[the remark this answers is not in the document]
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:28]: It's not really a term. So absolutely not :D

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+1. Install the `spryker/development` module.
+2. Open a terminal on your local machine.
+3. Navigate to your project directory.
```
[the remark this answers is not in the document]
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md]: applied

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+4. Run the following command in your project root:
+
+```bash Copy code
+vendor/bin/phpstorm-command-line-tools.sh
+```
+
+This command adds or updates an XML configuration file in PHPStorm’s settings directory, dedicated exclusively to Spryker-related configurations, without altering any other settings. The script currently supports Mac and Linux systems only.
```
[the remark this answers is not in the document]
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:38]: Yeap, you get a message from the console command that: "The configuration was successfully setup at {full path}".

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+3. Navigate to the root project directory.
+4. Run the following command in your project root:
+
+```bash Copy code
+vendor/bin/phpstorm-command-line-tools.sh
+```
+
+This command adds or updates an XML configuration file in PHPStorm’s settings directory, dedicated exclusively to Spryker-related configurations, without altering any other settings. The script currently supports Mac and Linux systems only.
+
+It is recommended to rerun this command periodically to apply the latest updates.
```
Comment (PhilinTv) [inline docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:42]: With development module updates or?

**Review by demkos — DISMISSED** [review]

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+To install the Spryker development tools configuration in PHPStorm:
+1. Install the [spryker/development@^3.39.0](https://github.com/spryker/development) module.
+2. Open a terminal on your local machine.
+3. Navigate to the root project directory.
+4. Run the following command in your project root:
+```bash Copy code
+vendor/bin/phpstorm-command-line-tools.sh
+```
+By default, the command installs with `--platform=docker`, configuring PHPStorm tools to run with the Spryker Docker SDK. To configure tools for local machine usage instead, specify `--platform=local`:
+```bash Copy code
```
Comment (andriitserkovnyi) [inline docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:41]: What do you mean by `copy code`? 

There's currently a button for that 
image

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+### Installation
+The installed commands within PHPStorm can be registered to run either as local machine commands or Spryker Docker SDK commands, depending on your configuration choice during installation. Consequently, ensure the appropriate environment is available (e.g., Spryker Docker SDK running for `docker` platform commands, or required tools installed locally for `local` platform commands).
+
+The script currently supports Mac systems only.
+
+**Prerequisite**:
+Ensure that `awk` is installed on your local machine, as it is required to run the commands directly from PHPStorm.
+
+To install the Spryker development tools configuration in PHPStorm:
```
Comment (andriitserkovnyi) [inline docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:32]: Does this script actually install something or just adds configuration?

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
+---
+title: PHPStorm Support
```
Comment (andriitserkovnyi) [inline docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:2]: What's the official name of the tools? Spryker Development Tools?

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md (hunk 9 above)
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:41]: I thought that it acts as a comment on the box :), nevermind

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md (hunk 10 above)
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:32]: Hmm, i'm not sure if i know what is the difference. It adds configuration only, but this adding is basically installs/enables functionality in PhpStorm.

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md (hunk 11 above)
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:2]: We are planning to have multiple PhpSTorm support tools (they are still in the backlog), so this is why i named this page like this, so we can have the upcoming tools here as well. 

So this script, enables the existing Spryker SDK tools in PhpStorm. I don't know what is the best way to express this.

File: docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+## PHPStorm Support
+Spryker provides support for tools that perform code style fixes, static analysis, and architecture checks, each accessible within PHPStorm’s External Tools panel. These tools are available at both module and project levels and can be executed directly by right-clicking within the project directory tree.
+
+The following tools are included:
+- `console code:sniff:style -f`: Applies coding style fixes (module folders only).
+- `console code:phpstan`: Runs static analysis (module folders only).
+- `console code:sniff:architecture`: Detects architecture rule violations (module folders only).
+
+### Installation
+The installed commands within PHPStorm can be registered to run either as local machine commands or Spryker Docker SDK commands, depending on your configuration choice during installation. Consequently, ensure the appropriate environment is available (e.g., Spryker Docker SDK running for `docker` platform commands, or required tools installed locally for `local` platform commands).
```
[the remark this answers is not in the document]
Comment (gerner-spryker) [reply docs/dg/dev/sdks/sdk/development-tools/phpstorm-support.md:25]: Spryker Docker is an environment that runs "inside" (on top of) your local machine. The installed software of your local machine is not the same as the installed software of the docker SDK environment.

Some shipped Spryker tools are actually not using the Spryker Docker environment, and they are expecting that you have installed development tools (eg: PHP) on your local machine directly (not inside the docker env).

Accordingly, this script asks you if you want the Spryker SDK tools using your local machine's installed software or inside the Spryker Docker environment (it's not the same set of Spryker SDK tools).

**Comment by gerner-spryker** [comment]
@andriitserkovnyi your changes look good to me

**Pull request merged**